File: server/src/main/scala/producer/message_generator/data_generators/FixedStringGenerator.scala
package producer.message_generator.data_generators

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class FixedStringGenerator()

object FixedStringGenerator:
    def fromPb(v: pb.FixedStringGenerator): FixedStringGenerator =
        FixedStringGenerator()

    def toPb(v: FixedStringGenerator): pb.FixedStringGenerator =
        pb.FixedStringGenerator()

File: ui/components/ui/ProducerSession/message-generator/data-generators/string-generator/fixed-string-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type FixedStringGenerator = {
  type: 'fixed-string-generator',
  string: string
};

export function fixedStringGeneratorFromPb(v: pb.FixedStringGenerator): FixedStringGenerator {
  return {
    type: 'fixed-string-generator',
    string: v.getString()
  };
}

export function fixedStringGeneratorToPb(v: FixedStringGenerator): pb.FixedStringGenerator {
  const message = new pb.FixedStringGenerator();
  message.setString(v.string);

  return message;
}
